zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/LearnerHandler.java

```diff
@@ -553,6 +554,10 @@ public void run() {
             }
             peerLastZxid = ss.getLastZxid();
 
+
+            ableToSkipTxn = QuorumZooKeeperServer.isSkipTxnAvailable(learnerProtocolVersion);
```

Comment:
`ableToSkipTxn` and `learnerProtocolVersion` are not declared.

Reply:
Same for this one, I will update this part.